### PR TITLE
Sphinx Action to Deploy on Github Pages

### LINKED ISSUE #38: Setup Doxygen or other code documentation
**Is your feature request related to a problem? Please describe.**

We should setup Doxygen or another code documentation suite ie. readthedocs to help generate dependency diagrams.

**Describe the solution you'd like**

Doxygen has a simple Github Action and one file that has to be added and can push to Github Pages...

**Describe alternatives you've considered**

Open to other ideas.

**Additional context**

Let's get a really large dynamic BFD (Big Friendly Diagram)

### PR DESCRIPTION
**Pull request by teadetime**
# Description
Action that deploys our documentation on github pages

Fixes #38 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+"""
+
+import os
+import sys
+
+sys.path.insert(0, os.path.abspath("../src"))
+autodoc_mock_imports = ["numpy", "pandas", "croniter"]
+
+project = "coin-test"
+copyright = "2022, Olin SCOPE"
```
Comment (intermezzio) [inline docs/conf.py:17]: 2023 now?

File: docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+templates_path = ["_templates"]
+exclude_patterns = []
+
+language = "python"
+
+# -- Options for HTML output -------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#options-for-html-output
+
+html_theme = "alabaster"
```
Comment (intermezzio) [inline docs/conf.py:35]: I see that this is the default option - any rationale for choosing this over the readthedocs theme (`"sphinx_rtd_theme"`) or something with a nicer-formatted sidebar like press (`"press"`)? For any other reviewers, the list of themes can be found [here](https://sphinx-themes.org/)

**Review by intermezzio — APPROVED** [review]
Left a few nits, can be updated in a future PR or changed now

File: .github/workflows/documentation.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - name: Install dependencies
+        run: |
+          pip install sphinx sphinx-autodoc-typehints
+      - run: pip install nox==2022.8.7
+      - run: pip install nox-poetry==1.0.1
+      - run: pip install poetry==1.3.2
```
Comment (eito-fis) [inline .github/workflows/documentation.yml:14]: Do we need to be installing nox and poetry?

**Pull request merged**